h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostUtils.java

```diff
@@ -75,21 +78,24 @@ public static DMatrix convertFrameToDMatrix(Key<DataInfo> dataInfoKey,
             chunks = VecUtils.getLocalChunkIds(f.anyVec());
         }
         final Vec weightVector = f.vec(weight);
-        int nRows = sumChunksLength(chunks, vec, weightVector);
+        final int[] nRowsByChunk = new int[chunks.length];
+        final long nRowsL = sumChunksLength(chunks, vec, weightVector, nRowsByChunk);
+        if (nRowsL > Integer.MAX_VALUE) {
```

Comment:
What if nRowsL overflows into negative ? Is this possible ? From my POV, this is not possible. Reasoning:

Max chunk len is `Integer.MAX_VALUE`. Max number of chunks here is also `INTEGER.MAX_VALUE - k`, where `k` is constant, typically 8. `Integer.MAX_VALUE` * `Integer.MAX_VALUE` is `4611686014132420609`. `Long.MAX_VALUE` is greater number - `9223372036854775807`.

Reply:
It is not possible, H2O will not let you store a dataset like that. Vec._espc is long[].